src/app/Components/Home/Home-Services/command-handler.service.ts

```diff
@@ -8,69 +8,62 @@ import { ErrorStateMatcher } from '@angular/material/core';
 import { AMY_CHAT_NAME, AMY_UNKNOWN_COMMAND_RESPONSE } from '../../../Constants/strings';
 import { OptionsService } from '../../../Services/options.service';
 import { ChatService } from './chat.service';
+import {BackendResolver} from '../../../Services/backendResolver.service';
+import {WebSocketSubject} from 'rxjs/webSocket';
 
 export class CommandErrorStateMatcher implements ErrorStateMatcher {
-  error = false;
-  isErrorState(control, form) {
-    return this.error;
-  }
+    error = false;
+    isErrorState(control, form) {
+        return this.error;
+    }
+}
+
+class ChatMessage {
+    msg: string;
 }
 
 @Injectable({
-  providedIn: 'root'
+    providedIn: 'root'
 })
 export class CommandHandlerService {
+    // shall the response of the KI backend system be read out loud
+    private readResponseState = false;
 
-  private readonly errorStateMatcher = new CommandErrorStateMatcher();
+    private readonly socket: WebSocketSubject<ChatMessage | string> = null;
 
-  // UUID -> id representing this conversation with the KI backend system
-  private uuid: string;
+    constructor(
+        private readonly databaseService: DatabaseService,
+        private readonly chat: ChatService,
+        private readonly options: OptionsService,
+        private readonly backendResolver: BackendResolver) {
 
-  // response of the KI backend system
-  response: string;
+        this.socket = new WebSocketSubject(this.generateWebSocketURL());
+        this.socket.subscribe((data) => {
+            this.chat.addMessage(AMY_CHAT_NAME[this.options.language], (data as ChatMessage).msg, this.readResponseState);
+        }, (error) => {
+            console.error(`WebSocket error: ${error}`);
+        }, () => {
+            console.error('WebSocket closed');
+        });
+    }
 
-  // shall the respnse of the KI backend system be read out loud
-  private readResponseState = false;
+    private generateWebSocketURL(): string {
+        const restURL = this.backendResolver.backendURL.getValue();
+        const port = 6661;
+        return restURL.replace(/http(s)?:\/\/(.*):([0-9]{1,})(\/.*)?/, `ws$1://$2:${port}`);
```

Comment:
this does not match an url without a port(80/443)

Reply:
Correct. I discussed the situation with @neumantm, it will not be possible to run the WebSocket server on port 6661 on our hosted instance anyway, so we need another solution

Reply:
server send events

Reply:
talk to @PassaChr about it